pkg/ipamclaimswebhook/podmutator.go

```diff
@@ -100,16 +113,41 @@ func (a *IPAMClaimsValet) Handle(ctx context.Context, request admission.Request)
 		}
 	}
 
-	newPrimaryNetworkIPAMClaimName, err := findNewPrimaryNetworkIPAMClaimName(ctx, a.Client, pod, vmName)
+	vmi, err := runningVMForPod(pod, vmName, a.Client)
+	if err != nil {
+		return admission.Errored(
+			http.StatusInternalServerError,
+			fmt.Errorf(
+				"failed to access the VMI running in pod %q: %w",
+				types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}.String(),
+				err,
+			),
+		)
+	}
+
+	primaryUDNInterface, err := findPrimaryUDNLogicalNetworkAttachment(ctx, a.Client, vmi)
 	if err != nil {
 		return admission.Errored(http.StatusInternalServerError,
 			fmt.Errorf("failed looking for primary user defined IPAMClaim name: %v", err))
 	}
-	if newPrimaryNetworkIPAMClaimName != "" {
+
+	if primaryUDNInterface != nil {
 		if newPod == nil {
 			newPod = pod.DeepCopy()
 		}
-		updatePodWithOVNPrimaryNetworkIPAMClaimAnnotation(newPod, newPrimaryNetworkIPAMClaimName)
+
+		primaryUDNIPRequests := primaryNetworkIPRequests(vmi, primaryUDNInterface.Name)
```

Comment:
Do you want to return if the Multus Default Network Annotation already lives, or do you want to calculate it every time?